src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/InternalUsersApiAction.java

```diff
@@ -116,8 +116,14 @@ protected void handlePut(RestChannel channel, final RestRequest request, final C
             }
         }
 
-        // if password is set, it takes precedence over hash
+        // Reject empty password
         final String plainTextPassword = securityJsonNode.get("password").asString();
+        if (plainTextPassword != null && plainTextPassword.isEmpty()){
+            badRequestResponse(channel, "Please specify a non-empty 'password' when creating a new internal user.");
+            return;
+        }
+
+        // if password is set, it takes precedence over hash
```

Comment:
Can you remove other checks for `isEmpty()` ?

Reply:
Done.

Reply:
Can you simplify the checks? we only need to check for password is not null and replace hash so that it takes precedence

Reply:
Done.